Missing dependencies in setup.py for running all the tests

**Describe the bug**
I installed a new fresh Satpy-development environment following the instructions:
https://satpy.readthedocs.io/en/stable/dev_guide/index.html#developer-s-guide

a handfull of tests fail due to missing packages/modules. these seem to be:
`python-eccodes`
`h5netcdf`
`xarray-datatree`
`skyfield`
`ephem`
`pint-xarray`
`astropy `

**To Reproduce**

Create new dev-environment as instructed and run

```python
pytest satpy/tests
```

**Expected behavior**
All tests should run once the dev-environment has been setup as per instructions
Missing dependencies should probably be added to:
https://github.com/pytroll/satpy/blob/main/setup.py#L38

**Actual results**
Example output from one of the test runs...
```
=============================================== short test summary info ================================================
FAILED satpy/tests/test_utils.py::TestGetSatPos::test_get_satpos_from_satname - ModuleNotFoundError: No module named 'skyfield'
FAILED satpy/tests/compositor_tests/test_viirs.py::TestVIIRSComposites::test_hncc_dnb_nomoonpha - ImportError: The 'ephem' library is required to calculate moon illumination fraction
FAILED satpy/tests/modifier_tests/test_parallax.py::TestParallaxCorrectionClass::test_correct_area_no_orbital_parameters - ModuleNotFoundError: No module named 'skyfield'
FAILED satpy/tests/modifier_tests/test_parallax.py::TestParallaxCorrectionSceneLoad::test_double_load - ModuleNotFoundError: No module named 'skyfield'
FAILED satpy/tests/modifier_tests/test_parallax.py::TestParallaxCorrectionSceneLoad::test_enhanced_image - ModuleNotFoundError: No module named 'skyfield'
FAILED satpy/tests/writer_tests/test_geotiff.py::TestGeoTIFFWriter::test_float_write_with_unit_conversion - ModuleNotFoundError: No module named 'pint_xarray'
```

**Environment Info:**
 - OS: macOS Ventura 13.3.1
 - Satpy Version: 0.42.2.dev12+g2aac9e4b6

--- a/doc/source/dev_guide/index.rst
+++ b/doc/source/dev_guide/index.rst
@@ -44,7 +44,7 @@ automatically reflected in the python environment. We highly recommend making
 a separate conda environment or virtualenv for development. For example, you
 can do this using conda_::
 
-  conda create -n satpy-dev python=3.8
+  conda create -n satpy-dev python=3.11
   conda activate satpy-dev
 
 .. _conda: https://conda.io/
@@ -80,7 +80,7 @@ libraries. If you want to run all Satpy tests you will need to install
 additional dependencies that aren't needed for regular Satpy usage. To install
 them run::
 
-    pip install -e .[tests]
+    pip install -e ".[tests]"
 
 Satpy tests can be executed by running::
 

--- a/setup.py
+++ b/setup.py
@@ -38,7 +38,8 @@ requires = ['numpy >=1.13', 'pillow', 'pyresample >=1.24.0', 'trollsift',
 test_requires = ['behave', 'h5py', 'netCDF4', 'pyhdf', 'imageio',
                  'rasterio', 'geoviews', 'trollimage', 'fsspec', 'bottleneck',
                  'rioxarray', 'pytest', 'pytest-lazy-fixture', 'defusedxml',
-                 's3fs']
+                 's3fs', 'python-eccodes', 'h5netcdf', 'xarray-datatree',
+                 'skyfield', 'ephem', 'pint-xarray', 'astropy']
 
 extras_require = {
     # Readers:

